doc/_ext/ceph_confval.py

```diff
@@ -1,12 +1,161 @@
+import io
+from typing import Any, Dict, List, Union
+
+from docutils.parsers.rst import directives
+from docutils.parsers.rst import Directive
+
 from sphinx.domains.python import PyField
 from sphinx.locale import _
+from sphinx.util import logging, status_iterator
 from sphinx.util.docfields import Field
 
+import jinja2
+import jinja2.filters
+import yaml
+
+
+logger = logging.getLogger(__name__)
+
+
+TEMPLATE = '''
+.. confval_option:: {{ opt.name }}
+{% if desc | length > 1 %}
+   {{ desc | wordwrap(70) | indent(3) }}
+{% endif %}
+   :type: ``{{opt.type}}``
+{%- if default %}
+  {%- if opt.type == 'size' %}
+   :default: ``{{ default | eval_size | filesizeformat(true) }}``
+  {%- elif opt.type in ('uint', 'int', 'float', 'secs') %}
+   :default: ``{{ default | readable_num(opt.type) }}``
+  {%- elif opt.type == 'millisecs' %}
+   :default: ``{{ default }}`` milliseconds
+  {%- elif opt.type == 'bool' %}
+   :default: ``{{ default | string | lower }}``
+  {%- else %}
+   :default: ``{{ default }}``
+  {%- endif -%}
+{%- endif %}
+{% if opt.versionchanged %}
+   .. versionchanged:: {{ opt.versionchanged }}
+{%- endif %}
+'''
+
+
+def eval_size(value) -> int:
+    try:
+        return int(value)
+    except ValueError:
+        times = dict(_K=1 << 10,
+                     _M=1 << 20,
+                     _G=1 << 30,
+                     _T=1 << 40)
+        for unit, m in times.items():
+            if value.endswith(unit):
+                return int(value[:-len(unit)]) * m
+        raise ValueError(f'unknown value: {value}')
+
+
+def readable_duration(value: str, typ: str) -> str:
+    try:
+        if typ == 'float':
+            v = float(value)
+        else:
+            v = int(value)
+        postfix = 'second' if v <= 1 else 'seconds'
```

Comment:
I think it should only use singular for exactly 1, i.e. saying 'that took 0 seconds' is correct

Reply:
fixed and repushed.

Reply:
thanks. TIL =)